### ISSUE TITLE
trace file not written to S3

### ISSUE THREAD

**Issue by kern3020** [nextflow-io/nextflow#2342]
## Bug report 

Nextflow can generate a trace file. The config file for this testcase is setup to do that.

    trace {
      enabled = true
      overwrite = true
      fields = "task_id,hash,native_id,process,tag,name,status,exit,module,container,cpus,time,disk,memory,attempt,submit,start,complete,duration,realtime,queue,%cpu,%mem,rss,vmem,peak_rss,peak_vmem,rchar,wchar,syscr,syscw,read_bytes,write_bytes"
      file = "s3://ag-ephemeral/jkern/1564-3/reports/trace.txt"
      raw = true
    }

Recently we noticed that it doesn't work with s3. 

[aside: there is a workaround using the -with-trace option to a local file and uploading it explicitly to s3. ]

### Expected behavior and actual behavior

Since the trace scope provides an s3 path, I expected it to be written there. 

### Steps to reproduce the problem

To demonstrate there is a hello world nextflow script.

    process sayHello {
        """
        sleep 30
        """
    }

and the associated nextflow.config file

    report {
        enabled = true
        overwrite = true
        file = "s3://ag-ephemeral/jkern/1564-3/reports/report.html"
    }

    timeline {
        enabled = true
        overwrite = true
        file = "s3://ag-ephemeral/jkern/1564-3/reports/timeline.html"
    }

    trace {
        enabled = true
        overwrite = true
        fields = "task_id,hash,native_id,process,tag,name,status,exit,module,container,cpus,time,disk,memory,attempt,submit,start,complete,duration,realtime,queue,%cpu,%mem,rss,vmem,peak_rss,peak_vmem,rchar,wchar,syscr,syscw,read_bytes,write_bytes"
        file = "s3://ag-ephemeral/jkern/1564-3/reports/trace.txt"
        raw = true
    }

    process {
        executor = "awsbatch"
        queue = "compute"
        container = "<...omitted...>.dkr.ecr.us-west-1.amazonaws.com/firefly37script:latest"
    }

    aws {
    acl = "BucketOwnerFullControl"
    }

### Program output 

```
bash-3.2$ nextflow run testcase.nf -w s3://ag-ephemeral/testing/work/
N E X T F L O W  ~  version 21.04.3
Launching `testcase.nf` [nasty_tesla] - revision: 292cda01f8
executor >  awsbatch (1)
[b3/6ec55d] process > sayHello [100%] 1 of 1 ✔
Completed at: 22-Sep-2021 15:19:04
Duration    : 5m 3s
CPU hours   : (a few seconds)
Succeeded   : 1
```

```
bash-3.2$ aws s3 ls s3://ag-ephemeral/jkern/1564-3/reports/
2021-09-22 15:19:04          0 
2021-09-22 15:19:04    2921905 report.html
2021-09-22 15:19:05     251866 timeline.html
bash-3.2$ 
```

.nextflow.log 

```
Sep-22 15:13:54.171 [main] DEBUG nextflow.cli.Launcher - $> nextflow run testcase.nf -w 's3://ag-ephemeral/testing/work/'
Sep-22 15:13:54.257 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 21.04.3
Sep-22 15:13:54.277 [main] INFO  nextflow.cli.CmdRun - Launching `testcase.nf` [nasty_tesla] - revision: 292cda01f8
Sep-22 15:13:54.289 [main] DEBUG nextflow.config.ConfigBuilder - Found config local: /Users/jkern/wip/eng-1564/nextflow.config
Sep-22 15:13:54.290 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/jkern/wip/eng-1564/nextflow.config
Sep-22 15:13:54.311 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `standard`
Sep-22 15:13:54.862 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; plugins-dir=/Users/jkern/.nextflow/plugins
Sep-22 15:13:54.863 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-amazon@1.0.5]
Sep-22 15:13:54.867 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins local root: .nextflow/plr/87881143a005e0d03979c7dc22790185
Sep-22 15:13:54.874 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Enabled plugins: []
Sep-22 15:13:54.875 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Disabled plugins: []
Sep-22 15:13:54.878 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode
Sep-22 15:13:54.895 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-amazon@1.0.5' resolved
Sep-22 15:13:54.903 [main] DEBUG nextflow.plugin.PluginUpdater - Starting plugin nf-amazon version: 1.0.5
Sep-22 15:13:54.903 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-amazon@1.0.5'
Sep-22 15:13:54.916 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-amazon@1.0.5
Sep-22 15:13:54.932 [main] DEBUG nextflow.file.FileHelper - > Added 'S3FileSystemProvider' to list of installed providers [s3]
Sep-22 15:13:54.977 [main] DEBUG nextflow.Session - Session uuid: e04e12b3-0f54-44f5-9110-861857552b19
Sep-22 15:13:54.977 [main] DEBUG nextflow.Session - Run name: nasty_tesla
Sep-22 15:13:54.978 [main] DEBUG nextflow.Session - Executor pool size: 12
Sep-22 15:13:54.989 [main] DEBUG nextflow.file.FileHelper - Creating a file system instance for provider: S3FileSystemProvider
Sep-22 15:13:55.000 [main] DEBUG nextflow.Global - Using AWS credential defined in `default` section in file: <...omitted...>
Sep-22 15:13:55.001 [main] DEBUG nextflow.file.FileHelper - AWS S3 config details: <...omitted...>
Sep-22 15:13:55.340 [main] WARN  c.a.a.p.i.BasicProfileConfigLoader - Your profile name includes a 'profile ' prefix. This is considered part of the profile name in the Java SDK, so you will need to include this prefix in your profile name when you reference this profile from your Java code.
Sep-22 15:13:55.340 [main] WARN  c.a.a.p.i.BasicProfileConfigLoader - Your profile name includes a 'profile ' prefix. This is considered part of the profile name in the Java SDK, so you will need to include this prefix in your profile name when you reference this profile from your Java code.
Sep-22 15:13:55.340 [main] WARN  c.a.a.p.i.BasicProfileConfigLoader - Your profile name includes a 'profile ' prefix. This is considered part of the profile name in the Java SDK, so you will need to include this prefix in your profile name when you reference this profile from your Java code.
Sep-22 15:13:55.340 [main] WARN  c.a.a.p.i.BasicProfileConfigLoader - Your profile name includes a 'profile ' prefix. This is considered part of the profile name in the Java SDK, so you will need to include this prefix in your profile name when you reference this profile from your Java code.
Sep-22 15:13:55.340 [main] WARN  c.a.a.p.i.BasicProfileConfigLoader - Your profile name includes a 'profile ' prefix. This is considered part of the profile name in the Java SDK, so you will need to include this prefix in your profile name when you reference this profile from your Java code.
Sep-22 15:13:55.340 [main] WARN  c.a.a.p.i.BasicProfileConfigLoader - Your profile name includes a 'profile ' prefix. This is considered part of the profile name in the Java SDK, so you will need to include this prefix in your profile name when you reference this profile from your Java code.
Sep-22 15:14:00.387 [main] DEBUG nextflow.cli.CmdRun - 
  Version: 21.04.3 build 5560
  Created: 21-07-2021 15:09 UTC (08:09 PDT)
  System: Mac OS X 11.5.2
  Runtime: Groovy 3.0.7 on OpenJDK 64-Bit Server VM 15.0.2+7
  Encoding: UTF-8 (UTF-8)
  Process: 7272@ag-admins-MacBook-Pro.local [192.168.21.64]
  CPUs: 12 - Mem: 16 GB (494 MB) - Swap: 3 GB (1.2 GB)
Sep-22 15:14:00.883 [main] WARN  com.amazonaws.util.Base64 - JAXB is unavailable. Will fallback to SDK implementation which may be less performant
Sep-22 15:14:00.888 [main] DEBUG nextflow.Session - Work-dir: s3://ag-ephemeral/testing/work [Mac OS X]
Sep-22 15:14:00.888 [main] DEBUG nextflow.Session - Script base path does not exist or is not a directory: /Users/jkern/wip/eng-1564/bin
Sep-22 15:14:00.914 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[AwsBatchExecutor]
Sep-22 15:14:00.923 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory
Sep-22 15:14:01.012 [main] DEBUG nextflow.Session - Session start invoked
Sep-22 15:14:01.014 [main] DEBUG nextflow.trace.TraceFileObserver - Flow starting -- trace file: /ag-ephemeral/jkern/1564-3/reports/trace.txt
Sep-22 15:14:01.221 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution
Sep-22 15:14:01.232 [main] DEBUG nextflow.Session - Workflow process names [dsl1]: sayHello
Sep-22 15:14:01.255 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: awsbatch
Sep-22 15:14:01.256 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'awsbatch'
Sep-22 15:14:01.257 [main] DEBUG nextflow.executor.Executor - [warm up] executor > awsbatch
Sep-22 15:14:01.268 [main] DEBUG nextflow.util.ThrottlingExecutor - Creating throttling executor with opts: nextflow.util.ThrottlingExecutor$Options(poolName:AWSBatch-executor, limiter:RateLimiter[stableRate=50.0qps], poolSize:60, maxPoolSize:60, queueSize:5000, maxRetries:10, keepAlive:1m, autoThrottle:true, errorBurstDelay:1s, rampUpInterval:100, rampUpFactor:1.2, rampUpMaxRate:1.7976931348623157E308, backOffFactor:2.0, backOffMinRate:0.0166666667, retryDelay:1s)
Sep-22 15:14:01.273 [main] DEBUG nextflow.util.ThrottlingExecutor - Creating throttling executor with opts: nextflow.util.ThrottlingExecutor$Options(poolName:AWSBatch-reaper, limiter:RateLimiter[stableRate=50.0qps], poolSize:60, maxPoolSize:60, queueSize:5000, maxRetries:10, keepAlive:1m, autoThrottle:true, errorBurstDelay:1s, rampUpInterval:100, rampUpFactor:1.2, rampUpMaxRate:1.7976931348623157E308, backOffFactor:2.0, backOffMinRate:0.0166666667, retryDelay:1s)
Sep-22 15:14:01.274 [main] DEBUG n.cloud.aws.batch.AwsBatchExecutor - Creating parallel monitor for executor 'awsbatch' > pollInterval=10s; dumpInterval=5m
Sep-22 15:14:01.294 [main] DEBUG nextflow.Global - Using AWS credential defined in `default` section in file: /Users/jkern/.aws/credentials
Sep-22 15:14:01.505 [main] DEBUG n.cloud.aws.batch.AwsBatchExecutor - [AWS BATCH] Executor options=AwsOptions(cliPath:null, storageClass:null, storageEncryption:null, remoteBinDir:null, region:null, maxParallelTransfers:16, maxTransferAttempts:1, delayBetweenAttempts:10s, fetchInstanceType:false, jobRole:null, volumes:[], awsCli:aws)
Sep-22 15:14:01.598 [main] DEBUG nextflow.script.ScriptRunner - > Await termination 
Sep-22 15:14:01.598 [main] DEBUG nextflow.Session - Session await
Sep-22 15:14:02.179 [AWSBatch-executor-1] DEBUG n.c.aws.batch.AwsBatchTaskHandler - [AWS BATCH] Found job definition name=nf-505086793999-dkr-ecr-us-west-1-amazonaws-com-firefly37script-latest:1; container=505086793999.dkr.ecr.us-west-1.amazonaws.com/firefly37script:latest
Sep-22 15:14:02.290 [AWSBatch-executor-1] DEBUG n.c.aws.batch.AwsBatchTaskHandler - [AWS BATCH] submitted > job=ff2b5c72-bf3d-4b6e-af90-5b8e5580b291; work-dir=s3://ag-ephemeral/testing/work/b3/6ec55dd8def5b9b89d3b33c3c34a26
Sep-22 15:14:02.290 [AWSBatch-executor-1] INFO  nextflow.Session - [b3/6ec55d] Submitted process > sayHello
Sep-22 15:19:01.675 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 1; name: sayHello; status: COMPLETED; exit: 0; error: -; workDir: s3://ag-ephemeral/testing/work/b3/6ec55dd8def5b9b89d3b33c3c34a26]
Sep-22 15:19:01.877 [main] DEBUG nextflow.Session - Session await > all process finished
Sep-22 15:19:01.926 [main] DEBUG nextflow.Session - Session await > all barriers passed
Sep-22 15:19:01.932 [main] DEBUG nextflow.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=1; failedCount=0; ignoredCount=0; cachedCount=0; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=30s; failedDuration=0ms; cachedDuration=0ms;loadCpus=0; loadMemory=0; peakRunning=1; peakCpus=1; peakMemory=0; ]
Sep-22 15:19:01.933 [main] DEBUG nextflow.trace.TraceFileObserver - Flow completing -- flushing trace file
Sep-22 15:19:02.026 [main] DEBUG nextflow.trace.ReportObserver - Flow completing -- rendering html report
Sep-22 15:19:02.051 [main] DEBUG nextflow.trace.ReportObserver - Execution report summary data:
  [{"cpuUsage":null,"process":"sayHello","mem":{"mean":4091904,"min":4091904,"q1":4091904,"q2":4091904,"q3":4091904,"max":4091904,"minLabel":"sayHello","maxLabel":"sayHello","q1Label":"sayHello","q2Label":"sayHello","q3Label":"sayHello"},"memUsage":null,"timeUsage":null,"vmem":{"mean":23523328,"min":23523328,"q1":23523328,"q2":23523328,"q3":23523328,"max":23523328,"minLabel":"sayHello","maxLabel":"sayHello","q1Label":"sayHello","q2Label":"sayHello","q3Label":"sayHello"},"reads":{"mean":49811,"min":49811,"q1":49811,"q2":49811,"q3":49811,"max":49811,"minLabel":"sayHello","maxLabel":"sayHello","q1Label":"sayHello","q2Label":"sayHello","q3Label":"sayHello"},"cpu":null,"time":{"mean":30004,"min":30004,"q1":30004,"q2":30004,"q3":30004,"max":30004,"minLabel":"sayHello","maxLabel":"sayHello","q1Label":"sayHello","q2Label":"sayHello","q3Label":"sayHello"},"writes":{"mean":165,"min":165,"q1":165,"q2":165,"q3":165,"max":165,"minLabel":"sayHello","maxLabel":"sayHello","q1Label":"sayHello","q2Label":"sayHello","q3Label":"sayHello"}}]
Sep-22 15:19:03.583 [main] DEBUG nextflow.trace.TimelineObserver - Flow completing -- rendering html timeline
Sep-22 15:19:04.404 [main] DEBUG nextflow.CacheDB - Closing CacheDB done
Sep-22 15:19:04.404 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-amazon@1.0.5'
Sep-22 15:19:04.404 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-amazon
Sep-22 15:19:04.412 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye
```

### Environment 

* Nextflow version: 21.04.3
* Java version:
openjdk version "15.0.2" 2021-01-19
OpenJDK Runtime Environment (build 15.0.2+7)
OpenJDK 64-Bit Server VM (build 15.0.2+7, mixed mode, sharing)
* Operating system: macOS
* Bash version: 3.2

**Comment by harper357** [comment]
I am also having this problem with Nextflow 22.04 when running it on AWS batch.

**Comment by justinjj24** [comment]
I am also having the same issue with my aws batch jobs

**Comment by pditommaso** [comment]
This has been solved via 8c643074 and 9926d15d. Include in version 22.10.0
